### PR TITLE
Fixes for SCP-firmware support

### PR DESCRIPTION
**Pull request by etienne-lms**
These changes apply to the SCP-firmware OP-TEE currently points to (v2.13.0). They'll need to be considered also when we'll upgrade to a more recent version of SCP-firmware, as proposed in P-R #6769.

### CODE REVIEW

**Review by jforissier — COMMENTED** [review]
- For "drivers: include stdbool.h from stpmic1_regulator.h":
  `Reviewed-by: Jerome Forissier <jerome.forissier@linaro.org>`
- For the other commits, with CI fixed...
  `Acked-by: Jerome Forissier <jerome.forissier@linaro.org>`

**Comment by etienne-lms** [comment]
Thanks for the review. I fixed the bug that made CI to fail. I'll apply the review tag if CI tests are happy.

File: core/lib/scmi-server/sub.mk [hunk 1]
```diff
@@ -167,6 +167,7 @@ endif
 endif
 
 cflags-lib-$(CFG_SCPFW_MOD_$4) += -DBUILD_HAS_MOD_$4
+scpfw-cmake-flags-y += -DCFG_SCPFW_MOD_$4=$(CFG_SCPFW_MOD_$4)
```
Comment (etienne-lms) [inline core/lib/scmi-server/sub.mk:175]: @vingu-linaro: considering your P-R #6769, the change above would deserve to factorize declaration of CMake directives in SCP-firmware as part of your [M-R #948](https://gitlab.arm.com/firmware/SCP-firmware/-/merge_requests/948): patch for _SCP-firmware/arch/none/optee/CMakeLists.txt_ (or maybe to apply in _SCP-firmware/product/optee/common/CMakeList.txt_):
[0001-product-optee-declare-CFG_SCPFW_MOD_xxx-CMake-direct.patch.txt](https://github.com/OP-TEE/optee_os/files/14995422/0001-product-optee-declare-CFG_SCPFW_MOD_xxx-CMake-direct.patch.txt)

Comment (vingu-linaro) [reply core/lib/scmi-server/sub.mk]: @etienne-lms:  Yes, I have added it into my SCP-firmware MR. I will push this new MR soon and will put your 2 other patches on top of this PR

Comment (etienne-lms) [reply core/lib/scmi-server/sub.mk]: Thanks.

**Comment by etienne-lms** [comment]
I have remove the change that creates CMake `CFG_SCPFW_MOD_<module-name>` directives and the one that enables `CFG_SCPFW_MOD_MSG_SMT` for plat-stm32mp1. Indeed SCP-firmware v2.13.0 is not ready for these changes.

File: core/lib/scmi-server/sub.mk [hunk 2]
```diff
@@ -30,6 +30,11 @@ scpfw-cmake-flags-y = -DSCP_FIRMWARE_SOURCE_DIR:PATH=$(scpfw-product)/fw \
 		      -DSCP_OPTEE_DIR:PATH=$(CURDIR) \
 		      -DCFG_CROSS_COMPILE=$(lastword $(CROSS_COMPILE_core))
 
+# CMake does not need to check the cross compilation toolchain since we do not
+# compile any source file with CMake, we only generate some SCP-firmware
+# files.
+scpfw-cmake-flags-y = -DCMAKE_C_COMPILER_WORKS=1
```
Comment (vingu-linaro) [inline core/lib/scmi-server/sub.mk]: should be 
scpfw-cmake-flags-y += -DCMAKE_C_COMPILER_WORKS=1
to not overwrite other cmake config
made the change in https://github.com/OP-TEE/optee_os/pull/6769

**Comment by etienne-lms** [comment]
@jforissier, I've applied your review tag on the commit you acked but I changed the other commit: @vingu-linaro noticed to _stddef.h_ was missing in the _stpmic1_regulator.h_.
Not a big change but I need to ask for review tags for commit "drivers: include standard header files from stpmic1_regulator.h"

**Comment by jforissier** [comment]
For "drivers: include standard header files from stpmic1_regulator.h":

`Reviewed-by: Jerome Forissier <jerome.forissier@linaro.org>`

**Comment by etienne-lms** [comment]
thanks. tag applied.

**Pull request merged**